pkg/storage/unified/migrations/README.md

```diff
@@ -0,0 +1,120 @@
+
+The migration system transfers resources from legacy SQL tables to Grafana's unified storage backend. It runs automatically during Grafana startup and validates data integrity after each migration.
+
+### Supported resources
+
+| Resource | API Group | Legacy table |
+|----------|-----------|--------------|
+| Folders | `folder.grafana.app` | `dashboard` |
+| Dashboards | `dashboard.grafana.app` | `dashboard` |
+| Library panels | `dashboard.grafana.app` | `library_element` |
+| Playlists | `playlist.grafana.app` | `playlist` |
+
+## Architecture
+
+```
+┌─────────────────────────────────────────────────────────────┐
+│                    ResourceMigration                         │
+│        (Orchestrates per-organization migration)             │
+└──────────────────────────┬──────────────────────────────────┘
+                           │
+       ┌───────────────────┼───────────────────┐
+       ▼                   ▼                   ▼
+  UnifiedMigrator      Validators         BulkProcess API
+  (Stream legacy     (Validate after      (Write to unified
+   resources)         migration)           storage)
+```
+
+### Components
+
+- **`service.go`**: Migration service entry point and registration
+- **`migrator.go`**: Core migration logic using streaming BulkProcess API
+- **`resource_migration.go`**: Per-organization migration execution
+- **`validator.go`**: Post-migration validation (CountValidator, FolderTreeValidator)
+- **`resources.go`**: Registry of migratable resource types
+
+## How migrations work
+
+### Migration flow
+
+1. Grafana starts and checks migration status in `unifiedstorage_migration_log` table
+2. For each organization, the migrator:
+   - Reads resources from legacy SQL tables
+   - Streams resources to unified storage via BulkProcess API
+   - Runs validators to verify data integrity
+3. Records migration result in `unifiedstorage_migration_log` table
+
+### Per-organization execution
+
+Migrations run independently for each organization using namespace format `org-{orgId}`.
+
+## Validators
+
+### CountValidator
+
+Compares resource counts between legacy SQL and unified storage. Accounts for rejected items during validation.
+
+### FolderTreeValidator
+
+Verifies folder parent-child relationships are preserved after migration.
+
+## Configuration
+
+To enable migrations, set the following in your Grafana configuration:
+
+```ini
+[unified_storage]
+disable_data_migrations = false
+```
+
+## Monitoring
+
+### Log messages
+
+Successful migration:
+
+```
+info: storage.unified.resource_migration Starting migration for all organizations
+info: storage.unified.resource_migration Migration completed successfully for all organizations
+```
+
+Failed migration:
+
+```
+error: storage.unified.resource_migration Migration validation failed
+```
+
+### Migration status
+
+Query the migration log table to check status:
+
+```sql
+SELECT * FROM unifiedstorage_migration_log WHERE migration_id LIKE '%folders-dashboards%';
```

Comment:
I think it would be nice to include a note about what `migration_id` is and how I can generalize the query based on other resource types.